### PR TITLE
Add translation of MAXSwerve-Java-Template

### PR DESCRIPTION
**Pull request by virtuald**
Tried to create a physics.py for this but... REV's simulation support only works in Java. Sad.

### CODE REVIEW

File: examples/maxswerve/constants.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+class NeoMotorConstants:
+    kFreeSpeedRpm = 5676
+
+
+class DriveConstants:
+    # Driving Parameters - Note that these are not the maximum capable speeds of
+    # the robot, rather the allowed maximum speeds
+    kMaxSpeedMetersPerSecond = 4.8
+    kMaxAngularSpeed = 2 * math.pi  # radians per second
```
Comment (auscompgeek) [inline examples/maxswerve/constants.py]:
```suggestion
    kMaxAngularSpeed = math.tau  # radians per second
```

File: examples/maxswerve/constants.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    kDrivingEncoderPositionFactor = (
+        kWheelDiameterMeters * math.pi
+    ) / kDrivingMotorReduction  # meters
+    kDrivingEncoderVelocityFactor = (
+        (kWheelDiameterMeters * math.pi) / kDrivingMotorReduction
+    ) / 60.0  # meters per second
+
+    kTurningEncoderPositionFactor = 2 * math.pi  # radian
+    kTurningEncoderVelocityFactor = (2 * math.pi) / 60.0  # radians per second
```
Comment (auscompgeek) [inline examples/maxswerve/constants.py]:
```suggestion
    kTurningEncoderPositionFactor = math.tau  # radian
    kTurningEncoderVelocityFactor = math.tau / 60.0  # radians per second
```

File: examples/maxswerve/subsystems/drivesubsystem.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        :param rateLimit:     Whether to enable rate limiting for smoother control.
+        """
+
+        xSpeedCommanded = xSpeed
+        ySpeedCommanded = ySpeed
+
+        if rateLimit:
+            # Convert XY to polar for rate limiting
+            inputTranslationDir = math.atan2(ySpeed, xSpeed)
+            inputTranslationMag = math.sqrt(math.pow(xSpeed, 2) + math.pow(ySpeed, 2))
```
Comment (auscompgeek) [inline examples/maxswerve/subsystems/drivesubsystem.py]:
```suggestion
            inputTranslationMag = math.hypot(xSpeed, ySpeed)
```

Comment (auscompgeek) [inline examples/maxswerve/subsystems/drivesubsystem.py]:
```suggestion
            inputTranslationMag = math.hypot(xSpeed, ySpeed)
```

File: examples/maxswerve/subsystems/drivesubsystem.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+            ChassisSpeeds.fromFieldRelativeSpeeds(
+                xSpeedDelivered,
+                ySpeedDelivered,
+                rotDelivered,
+                Rotation2d.fromDegrees(self.gyro.getAngle()),
+            )
+            if fieldRelative
+            else ChassisSpeeds(xSpeedDelivered, ySpeedDelivered, rotDelivered)
+        )
+        SwerveDrive4Kinematics.desaturateWheelSpeeds(
```
Comment (auscompgeek) [inline examples/maxswerve/subsystems/drivesubsystem.py]: Doesn't desaturate return a tuple?

File: examples/maxswerve/swerveutils.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+def wrapAngle(angle: float) -> float:
+    """Wraps an angle until it lies within the range from 0 to 2*PI (exclusive).
+
+    :param angle: The angle (in radians) to wrap.  Can be positive or negative and can lie multiple wraps outside the output range.
+
+    :returns: An angle (in radians) from 0 and 2*PI (exclusive).
+    """
+
+    twoPi = math.pi * 2
```
Comment (auscompgeek) [inline examples/maxswerve/swerveutils.py]: math.tau

File: examples/maxswerve/swerveutils.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    stepDirection = math.copysign(target - current, 1)
+    difference = abs(current - target)
+
+    if difference <= stepsize:
+        return target
+    elif difference > math.pi:  # does the system need to wrap over eventually?
+        # handle the special case where you can reach the target in one step while also wrapping
+        if (
+            current + 2 * math.pi - target < stepsize
+            or target + 2 * math.pi - current < stepsize
```
Comment (auscompgeek) [inline examples/maxswerve/swerveutils.py]:
```suggestion
            current + math.tau - target < stepsize
            or target + math.tau - current < stepsize
```

File: examples/maxswerve/swerveutils.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+def angleDifference(angleA: float, angleB: float) -> float:
+    """Finds the (unsigned) minimum difference between two angles including calculating across 0.
+
+    :param angleA: An angle (in radians).
+    :param angleB: An angle (in radians).
+
+    :returns: The (unsigned) minimum difference between the two angles (in radians).
+    """
+    difference = abs(angleA - angleB)
+    return (2 * math.pi) - difference if difference > math.pi else difference
```
Comment (auscompgeek) [inline examples/maxswerve/swerveutils.py]:
```suggestion
    return math.tau - difference if difference > math.pi else difference
```

File: examples/maxswerve/subsystems/drivesubsystem.py (hunk 4 above)
Comment (virtuald) [reply examples/maxswerve/subsystems/drivesubsystem.py]: Oh that's an interesting difference.

**Pull request merged**